prov/psm2/src/psmx2.h

```diff
@@ -1074,6 +1075,21 @@ static inline void psmx2_progress_all(struct psmx2_fid_domain *domain)
 	psmx2_unlock(&domain->trx_ctxt_lock, 1);
 }
 
+/*
+ * There is a limitation in PSM2 AM implementation that can cause significant
+ * delay if more than 811 AM requests are enqueued in a row without progress
+ * calls being made in between. As a workaround, call this function after each
+ * AM request is enqueued whenever possible.
+ */
+#define PSMX2_AM_POLL_INTERVAL	800
+static inline void psmx2_am_poll(struct psmx2_trx_ctxt *trx_ctxt)
+{
+	if (OFI_UNLIKELY(++trx_ctxt->am_poll_count > PSMX2_AM_POLL_INTERVAL)) {
+		trx_ctxt->am_poll_count = 0;
```

Comment:
Is am_poll_count thread safe, or does it really matter?  The reset to 0 may be lost, but I don't know if that could cause a real problem.

Reply:
it's not thread safe (by design) and yes it doesn't matter :)